File: GXpert/GXpert.Web/Modules/Attendance/AssignmentAttempt/AssignmentAttemptColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Attendance.Columns;

[ColumnsScript("Attendance.AssignmentAttempt")]
[BasedOnRow(typeof(AssignmentAttemptRow), CheckNames = true)]
public class AssignmentAttemptColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string AssignmentTitle { get; set; }
    public string StudentPrn { get; set; }
    [EditLink]
    public string FileUploaded { get; set; }
    public string EStatus { get; set; }
    public string TeacherPrn { get; set; }
    public string PlayListTitle { get; set; }
    public string ActivationDeviceId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/AssignmentAttemptColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssignmentAttemptRow } from "./AssignmentAttemptRow";

export interface AssignmentAttemptColumns {
    Id: Column<AssignmentAttemptRow>;
    AssignmentTitle: Column<AssignmentAttemptRow>;
    StudentPrn: Column<AssignmentAttemptRow>;
    FileUploaded: Column<AssignmentAttemptRow>;
    EStatus: Column<AssignmentAttemptRow>;
    TeacherPrn: Column<AssignmentAttemptRow>;
    PlayListTitle: Column<AssignmentAttemptRow>;
    ActivationDeviceId: Column<AssignmentAttemptRow>;
    InsertDate: Column<AssignmentAttemptRow>;
    InsertUserId: Column<AssignmentAttemptRow>;
    UpdateDate: Column<AssignmentAttemptRow>;
    UpdateUserId: Column<AssignmentAttemptRow>;
    IsActive: Column<AssignmentAttemptRow>;
}

export class AssignmentAttemptColumns extends ColumnsBase<AssignmentAttemptRow> {
    static readonly columnsKey = 'Attendance.AssignmentAttempt';
    static readonly Fields = fieldsProxy<AssignmentAttemptColumns>();
}